piet-web/src/lib.rs

```diff
@@ -270,45 +270,44 @@ impl RenderContext for WebRenderContext<'_> {
         let canvas = element.dyn_into::<HtmlCanvasElement>().unwrap();
         canvas.set_width(width as u32);
         canvas.set_height(height as u32);
-        let mut buf = match format {
-            // Discussion topic: if buf were mut here, we could probably avoid this clone.
-            // See https://github.com/rustwasm/wasm-bindgen/issues/2364 for the issue.
-            ImageFormat::RgbaSeparate => buf.to_vec(),
+        let mut new_buf: Vec<u8>;
```

Comment:
I'm confused that this will compile when this value isn't initialized in all branches?

Reply:
the compiler only checks branches where `new_buf` is used, and all of those are handled properly.  (I think thats how it works)